homeassistant/components/motionmount/number.py

```diff
@@ -0,0 +1,75 @@
+"""Support for MotionMount numeric control."""
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import MotionMountCoordinator
+from .entity import MotionMountEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Vogel's MotionMount from a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    unique_id = format_mac(coordinator.mm.mac.hex())
+
+    async_add_entities(
+        [
+            MotionMountExtension(coordinator, unique_id),
+            MotionMountTurn(coordinator, unique_id),
+        ]
+    )
+
+
+class MotionMountExtension(MotionMountEntity, NumberEntity):
+    """The target extension position of a MotionMount."""
+
+    _attr_native_max_value = 100
+    _attr_native_min_value = 0
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _attr_translation_key = "motionmount_extension"
+
+    def __init__(self, coordinator: MotionMountCoordinator, unique_id: str) -> None:
+        """Initialize Extension number."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{unique_id}-extension"
+        self._attr_native_value = self.coordinator.data["extension"]
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        self._attr_native_value = self.coordinator.data["extension"]
+        self.async_write_ha_state()
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set the new value for extension."""
+        await self.coordinator.mm.set_extension(int(value))
+
+
+class MotionMountTurn(MotionMountEntity, NumberEntity):
```

Comment:
Are you going to add more entities? Maybe take a look at the entity descriptions. This way you create 1 generic NumberEntity, and then use descriptions to make it specific for that one.

Reply:
I couldn't find any documentation about the entity descriptions. I'm going to make more entities, but not number entities. There will be 1 binary sensor, 1 select and 1 sensor. Will it then still be useful to use the entity description?

Reply:
In that case I think this is fine. It could make the code a bit cleaner but that's no requirement for this PR imo. (Checkout youtube/sensor.py for an example for how the entity descriptions work). But I'll leave that up to you.